dataflow/src/domain/mod.rs

```diff
@@ -1566,6 +1574,80 @@ impl Domain {
         }
     }
 
+    fn handle_recovery(&mut self) {
+        let node_info: Vec<_> = self.nodes
+            .iter()
+            .map(|(index, node)| {
+                let n = node.borrow();
+                (index.clone(), n.global_addr(), n.is_transactional())
```

Comment:
I don't think you need the `clone` here, because `index` is `Copy`. I think you can just do `*index`, or match against `&index` in the map pattern.

Reply:
Fixed! Didn't need the `.clone` at all, like you said.